### PR TITLE
Run3hcx378A Provide a way to avoid Digitization for the ZDC detector

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Provide a way to avoid Digitization for the ZDC detector

#### PR validation:

By using the following lines in the step2 of the workflow of 12834.0, the digitization of ZDC was switched off:
from Configuration.Eras.Era_Run3_2024_cff import Run3_2024
from Configuration.Eras.Modifier_zdcNoDigi_cff import zdcNoDigi
process = cms.Process('HLT',Run3_2024,zdcNoDigi)

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
 
Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch Please approve this. It is required by the ZDC people

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@antoniovilela,  @mandrenguyen Please approve this PR

**Comment by antoniovilela** [comment]
+1

File: SimCalorimetry/HcalSimProducers/python/hcalUnsuppressedDigis_cfi.py [hunk 1]
```diff
@@ -64,3 +65,7 @@
 # remove HE processing for phase 2, completely put in HGCal land
 from Configuration.Eras.Modifier_phase2_hgcal_cff import phase2_hgcal
 phase2_hgcal.toModify(hcalSimBlock, killHE = cms.bool(True) )
+
+# remove HE processing for phase 2, completely put in HGCal land
```
Comment (perrotta) [inline SimCalorimetry/HcalSimProducers/python/hcalUnsuppressedDigis_cfi.py:69]: This comment does not refer to the current case

File: SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc [hunk 2]
```diff
@@ -30,7 +30,7 @@
 #include <boost/foreach.hpp>
 #include <memory>
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (perrotta) [inline SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc:33]: This should be commented back

**Review by perrotta — COMMENTED** [review]
@bsunanda @hjbossi I'm noticing this PR only now. Besides these minor fixes, which are needed (also for a possible backport PR): is this modifier also expected to be used to switch off ZDC digitization also in Phase2 workflows? See for example the comment in https://github.com/cms-sw/cmssw/pull/46537#discussion_r1844490625

File: SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc (hunk 2 above)
Comment (civanch) [reply SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc:33]: There is #46708 with the fix.

Comment (perrotta) [reply SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc:33]: > There is #46708 with the fix.

Please either profit of that PR, or make a new one dedicated, to fix also the comment line as [above](https://github.com/cms-sw/cmssw/pull/46570#discussion_r1844930269)

**Pull request merged**